src/azul/service/repository_controller.py

```diff
@@ -66,6 +75,14 @@ def download_file(self,
             assert file_version is not None
             assert file_name is not None
 
+        # To avoid a `416 Range Not Satisfiable` response to curl clients
+        # using the `--continue-at -` option, return an empty 206 response
+        # if the requested byte range starts at the end of the file.
+        if start_byte is not None and start_byte == file_size:
```

Comment:
In PyCharm `file_size` gives a reference before assignment warning. You might want to extract the variable elsewhere to remove the warning or include the conditional in the block were the variable gets assigned.

Reply:
Nice catch. Since `start_byte` and `file_size` are only needed to see if this is a range request for a completed file, I think I'll opt for a boolean that can be set in the `if` condition instead of pre-initializing two variable just so they can be compared outside the block.